lib/utils/datahub.js

```diff
@@ -95,7 +98,40 @@ class DataHub extends EventEmitter {
     this._debugMsg('Calling source upload with spec')
     this._debugMsg(spec)
 
+
+    const token = await this._authz('source')
+    const res = await this._fetch('/source/upload', token, {
+      method: 'POST',
+      body: spec
+    })
+
+    if (res.status === 200) {
+      const out = await res.json()
+      this._debugMsg(out)
+      return out
+    }
+    throw new Error(responseError(res))
+  }
+
+  async pushFlow(url){
+    const tmpdir = tmp.dirSync()
+    try {
+      await Git.Clone(url, tmpdir.name)
+    } catch (err) {
+      throw new Error(`Failed to clone from ${url}`)
+    }
+    let spec = {}
+    try {
```

Comment:
Why do you need to load the yaml? Are you just to push the spec directly? In which case you could push it directly as a bytestream (this would save parsing yaml)? Up to you on this one as not a big deal to load yaml really.